core/eth/reader.go

```diff
@@ -916,3 +935,22 @@ func (t *Reader) GetRelayURLs(ctx context.Context) (map[uint32]string, error) {
 
 	return res, nil
 }
+
+func (t *Reader) GetDisperserAddress(ctx context.Context, disperserID uint32) (gethcommon.Address, error) {
+	registry := t.bindings.DisperserRegistry
+	if registry == nil {
+		return gethcommon.Address{}, errors.New("disperser registry not deployed")
+	}
+
+	address, err := registry.DisperserKeyToAddress(
+		&bind.CallOpts{
+			Context: ctx,
+		},
+		disperserID)
+
+	if err != nil {
+		return gethcommon.Address{}, fmt.Errorf("failed to get disperser address: %w", err)
+	}
+
```

Comment:
I don't think the contract will return `err` if the `disperserID` doesn't exist. It will probably return zero address. 
We need to explicitly check this case